pymc_marketing/mmm/evaluation.py

```diff
@@ -0,0 +1,238 @@
+        "mae": mean_absolute_error,
+        "nmae": nmae,
+        "mape": mean_absolute_percentage_error,
+    }
+
+    if metrics_to_calculate is None:
+        metrics_to_calculate = list(metric_functions.keys())
+    else:
+        invalid_metrics = set(metrics_to_calculate) - set(metric_functions.keys())
+        if invalid_metrics:
+            raise ValueError(
+                f"Invalid metrics: {invalid_metrics}. "
+                f"Valid options are: {list(metric_functions.keys())}"
+            )
+
+    results = {}
+    for metric in metrics_to_calculate:
+        metric_values = np.array(
+            [
+                metric_functions[metric](
+                    y_true, y_pred[:, i]
+                )  # Calculate along date dimension
+                for i in range(y_pred.shape[1])
+            ]
+        )
+        results[metric] = metric_values
+
+    return results
+
+
+def summarize_metric_distributions(
+    metric_distributions: dict[str, np.ndarray] | None = None,
+    hdi_prob: float = 0.94,
+) -> dict[str, dict[str, float]]:
+    """Summarize metric distributions with point estimates and HDIs.
+
+    Parameters
+    ----------
+    metric_distributions : dict of str to np.ndarray
+        Dictionary of metric distributions as returned by calculate_metric_distributions.
+    hdi_prob : float, optional
+        The probability mass of the highest density interval. Defaults to 0.94.
+
+    Returns
+    -------
+    dict of str to dict
+        A dictionary containing summary statistics for each metric.
+        List of summary statistics calculated for each metric:
+            * `mean`: Mean of the metric distribution.
+            * `median`: Median of the metric distribution.
+            * `std`: Standard deviation of the metric distribution.
+            * `min`: Minimum value of the metric distribution.
+            * `max`: Maximum value of the metric distribution.
+            * `hdi_lower`: Lower bound of the Highest Density Interval.
+            * `hdi_upper`: Upper bound of the Highest Density Interval.
+    Sets
+    ----
+    self.metric_summaries : dict of str to dict
+        Stores the calculated metric summaries as an instance attribute.
+    """
+    if metric_distributions is None:
+        raise ValueError(
+            "Metric distributions have not been calculated.\
+            Call `calculate_metric_distributions` first."
+        )
+
+    metric_summaries = {}
+    for metric, distribution in metric_distributions.items():
+        hdi = az.hdi(distribution, hdi_prob=hdi_prob)
+        metric_summaries[metric] = {
+            "mean": np.mean(distribution),
+            "median": np.median(distribution),
+            "std": np.std(distribution),
+            "min": np.min(distribution),
+            "max": np.max(distribution),
+            f"{hdi_prob:.0%}_hdi_lower": hdi[0],
+            f"{hdi_prob:.0%}_hdi_upper": hdi[1],
+        }
+
+    return metric_summaries
+
+
+def evaluate_model(
```

Comment:
I would rename this method to something more explicitly like: `compute_summary_metrics` (or similar) as it is closer to what is doing, what do you think?

Reply:
Yep, good idea - thanks!